scripts/configurator/parser/cmake_var.py

```diff
@@ -0,0 +1,40 @@
+from typing import Dict, Any
+
+def generate_cmake_var(type :str, cmake_var_str):
+   "factory function that associates the correct type of variable"
+   cmake_type_str_to_class = {
+      "BOOL" : cmake_var_bool,
+      "INTERNAL" : cmake_var_generic,
+      "FILEPATH": cmake_var_generic,
+      "STRING" : cmake_var_generic,
+   }
+   class_obj = cmake_type_str_to_class[type]
+   return class_obj(cmake_var_str)
+
+
+class cmake_var_generic:
+   "generic type of cmake_var, used when we don't need particular conversions"
+   def __init__(self, value: str) -> None:
+      self.value: Any = value
+
+   def serialize(self) -> str:
+      return self.value
+
+class cmake_var_bool(cmake_var_generic):
+
+   NORMALIZED_BOOLS: Dict[str, bool] = {
+      "ON" : True,
+      "OFF": False,
+      "1" : True,
+      "0" : False,
+      "TRUE": True,
+      "FALSE": False,
+   }
+
+   def __init__(self, value: str) -> None:
+      self.value = self.NORMALIZED_BOOLS[value]
+
+   def serialize(self) -> str:
+      return "ON" if self.value else "OFF"
+
+cmake_var = cmake_var_generic | cmake_var_bool
```

Comment:
Can you explain to be how this syntax works? It doesn't look like a linter thing. It looks like a type alternative that I saw only in pure functional languages like Haskell. I didn't know you could do things like that in Python.

Btw, if you use inheritance, you don't use type alternatives like that. You just use the base class, which should be simply called `cmake_var`. That's the whole point of polymorphism. Type alternatives are good when the types are NOT related. It's either one or the other. It doesn't make sense like that.

Reply:
Ok, that makes sense. `X | Y` is just syntactic sugar for `Union[X, Y]`. I think it's also used in typescript.